backend/database.go

```diff
@@ -0,0 +1,86 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"github.com/go-redis/redis/v8"
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate mockery --name=Database
+// Database struct contains address(String), password(string), connection, ctx.
+type Database struct {
+	address    string
+	password   string
+	connection *redis.Client
+	ctx        context.Context
+}
+
+// NewDatabaseConnection returns a Database structure using the given arguments.
+func NewDatabaseConnection(address, password string) Database {
+	return Database{
+		address:  address,
+		password: password,
+		ctx:      context.Background(),
+	}
+}
+
+// Connect returns an error if connection already has been initialized, otherwise creates a new connection.
+func (d *Database) Connect() error {
+	if d.connection != nil {
+		log.Info("Connection to database is already initialized")
+		return nil
+	}
+
+	d.connection = redis.NewClient(&redis.Options{
+		Addr:     d.address,
+		Password: d.password,
+		DB:       0,
+	})
+
+	if err := d.connection.Ping(d.ctx).Err(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// InsertToDB adds a database entry using provided key(String) and value(String).
+func (d Database) InsertToDB(key string, value string) error {
```

Comment:
You're already calling this method on type `Database` - consider renaming methods to avoid repetitions like:

db := Database{}
**db**.InsertTo**DB**


imo just `Insert()` would be enough, cause you always call it on `Database` type - think this way - where else could you insert data than `ToDB`?